1-python-question.py

```diff
@@ -23,13 +23,50 @@
     "applicants": ["Amy", "Ally", "David", "Brendan", "Zoho"]
 }
 
+def CheckBannedVisitors(applicant):
+    for banned_visitor in banned_visitors:
+        return banned_visitor == applicant
 
-def processRequest(request):
-    # Your code here
-    return
+def CheckSuccessfulApplicants(applicant):
+    for banned_visitor in banned_visitors:
+        return banned_visitor != applicant
 
+def Ticket(applicant):
+    membership_status = (applicant in member_status) and (member_status[applicant]==True)
+    if membership_status == True:
+        price = 5
+    else:
+        price = 3.50
+    ticket = {
+        "name": applicant,
+        "membershipStatus": membership_status,
+        "price": price,
+    }
+    return ticket
 
-print(processRequest(request))
+def ProcessRequest(request):
+    try:
+        if request["applicants"] == []:
+            raise ValueError("No applicants found!")
+        successful_applicants = list(filter(CheckSuccessfulApplicants, request['applicants']))
+        banned_applicants = list(filter(CheckBannedVisitors, request['applicants']))
+        total_cost = 0
+        ticket_list = []
+        for applicant in successful_applicants:
+            ticket = Ticket(applicant)
+            total_cost += ticket["price"]
+            ticket_list.append(ticket)
+        result = {
+                "successfulApplicants" : successful_applicants,
+                "bannedApplicants" : banned_applicants,
+                "totalCost": total_cost,
+                "tickets": ticket_list,
+        }
+        return result
+    except ValueError as err:
+        return err
```

Comment:
when you raise an error in the method, it cannot be caught by the same method as the exception is throw out of the method. Also the question does not specify that you have to catch the error.